WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/DashboardDomainRegistrationCardCell.swift

```diff
@@ -0,0 +1,110 @@
+                title: Strings.title,
+                description: Strings.content,
+                hideThis: Strings.hideThis,
+                source: Strings.source
+            ),
+            onViewTap: onViewTap,
+            onHideThisTap: onHideThisTap
+        )
+    }()
+
+    // MARK: - User Interaction
+
+    private func cardTapped() {
+        guard let props = makeUnwrappedProperties() else {
+            return
+        }
+        WPAnalytics.track(.domainCreditRedemptionTapped)
+        DomainsDashboardCoordinator.presentDomainsSuggestions(
+            in: props.presentingViewController,
+            source: Strings.source,
+            blog: props.blog
+        )
+    }
+
+    private func hideCardTapped() {
+        guard let props = makeUnwrappedProperties() else {
+            return
+        }
+        let service = BlogDashboardPersonalizationService(siteID: props.siteID.intValue)
+        service.setEnabled(false, for: .domainRegistration)
+    }
+
+    // MARK: - Constants
+
+    private static var hasLoggedDomainCreditPromptShownEvent: Bool = false
+
+    private enum Strings {
+        static let title = NSLocalizedString(
+            "Register Domain",
+            comment: "Action to redeem domain credit."
+        )
+        static let content = NSLocalizedString(
+            "All WordPress.com plans include a custom domain name. Register your free premium domain now.",
+            comment: "Information about redeeming domain credit on site dashboard."
+        )
+        static let hideThis = NSLocalizedString(
+            "domain.dashboard.card.menu.hide",
+            value: "Hide this",
+            comment: "Title for a menu action in the context menu on the Jetpack install card."
+        )
+        static let source = "domain_registration_dashboard_card"
+    }
+
+    // MARK: - View Lifecycle
+
+    override func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        guard !Self.hasLoggedDomainCreditPromptShownEvent else {
+            return
+        }
+        WPAnalytics.track(WPAnalyticsStat.domainCreditPromptShown)
+        Self.hasLoggedDomainCreditPromptShownEvent = true
+    }
+
+    // MARK: - Helpers
+
+    private func makeUnwrappedProperties() -> Unwrapped? {
+        return Unwrapped(presentingViewController: presentingViewController, blog: blog)
+    }
+
+    // MARK: - Supporting Types
+
+    private struct Unwrapped {
+        let presentingViewController: BlogDashboardViewController
+        let blog: Blog
+        let siteID: NSNumber
+
+        init?(presentingViewController: BlogDashboardViewController?,
+             blog: Blog?) {
+            guard let presentingViewController,
+                  let blog,
+                  let siteID = blog.dotComID else {
+                return nil
+            }
+            self.presentingViewController = presentingViewController
+            self.blog = blog
+            self.siteID = siteID
+        }
+    }
+}
```

Comment:
This feels like a code smell. It seems this class requires all three of these properties as non-optional to be functional but yet it has them as optionals. It might be pointing to a non-suitable inheritance or an initializer.

Reply:
This `struct` is just for convenience to avoid writing this code everytime I want to use those properties:
```swift
            guard let presentingViewController,
                  let blog,
                  let siteID = blog.dotComID else {
                return nil
            }
```
So instead of the code above, I can write just the following:
```swift
guard let props = Unwrapped(presentingViewController: presentingViewController, blog: blog) else {
   return
}
// Usage:
// props.presentingViewController
// props.siteID
```

Reply:
The guard-let unwrapping was used in `cardTapped`and `hideCardTapped`. I understand this approach is unconventional but it was convenient to use that type instead of unwrapping all properties multiple times.